src/nixpacks/plan/generator.rs

```diff
@@ -207,7 +207,9 @@ impl<'a> NixpacksBuildPlanGenerator<'a> {
                 img => Some(img.to_owned()),
             };
         }
-
+        if NodeProvider::uses_canvas(app) {
+            start_phase.cmd = Some(format!("for pkg in $(ls /nix/store | grep \"util-linux-.*-lib$\"); do export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:\"/nix/store/$pkg/lib\"; done && {}", start_phase.cmd.unwrap_or_default()));
+        }
```

Comment:
The generator (or nixpacks in general) shouldn't know anything specific about the providers. This should be moved to the node provider start command.

Reply:
Deployment will crash if someone has `Procfile` in their repo.

Reply:
Hmm then need to find a better way to handle Procfile start commands. Or handle util-linux natively in the base image. The node provider should not be imported and used like this. This doesn't seem like a bad thing to have in the base. Do you know how much larger it makes the image?

<img width="932" alt="image" src="https://user-images.githubusercontent.com/3044853/172722651-0fa86334-65d9-4f0a-b9cc-8cc369d26f51.png">

The providers currently have a way of specifying environment variables for the container. We could also expand that functionality so that they can add dynamic things like this?

Reply:
As mentioned in #183 letting providers read/edit `environment.nix` directly would be better IMO. ([link](https://discourse.nixos.org/t/node2nix-issues/10762/2))

Reply:
The `environment.nix` file doesn't exist when the provider lifecycle methods are run so there is nothing to read/edit yet. I do think though that the providers need a way to inject various Nix snippets into the environment that gets run. Just need to land on an abstraction that will work in the majority of cases.

Reply:
Even just being able to edit `environment.nix` isn't enough. This is an example that would write the library path to a `env.sh` file under the default Nix profile

```nix
{ }:

let pkgs = import <nixpkgs> { };
in with pkgs;
let
  APPEND_LIBRARY_PATH = "${lib.makeLibraryPath [ libGL libuuid ]}";
  myProfile = writeText "env" ''
    export LD_LIBRARY_PATH="${APPEND_LIBRARY_PATH}:$LD_LIBRARY_PATH"
  '';

in buildEnv {
  name = "env";
  paths = [
    (runCommand "env" { } ''
      mkdir -p $out/etc/profile.d
      cp ${myProfile} $out/etc/profile.d/env.sh
    '')

    nodejs
  ];
}
```

But we would still need to somehow load that so those variables are available at runtime.

I think for the `LD_LIBRARY_PATH` variable specifically, we should have a `setup_phase.set_libraries` function that would set everything up automatically.

Reply:
I think this makes adding the libraries available much easier https://github.com/railwayapp/nixpacks/pull/197